src/protocols/gamma_control.rs

```diff
@@ -0,0 +1,237 @@
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::Read;
+
+use smithay::reexports::wayland_protocols_wlr;
+use smithay::reexports::wayland_server::backend::{ClientId, ObjectId};
+use smithay::reexports::wayland_server::protocol::wl_output::WlOutput;
+use smithay::reexports::wayland_server::{
+    Client, DataInit, Dispatch, DisplayHandle, GlobalDispatch, New, Resource,
+};
+use wayland_protocols_wlr::gamma_control::v1::server::{
+    zwlr_gamma_control_manager_v1, zwlr_gamma_control_v1,
+};
+use zwlr_gamma_control_manager_v1::ZwlrGammaControlManagerV1;
+use zwlr_gamma_control_v1::ZwlrGammaControlV1;
+
+const VERSION: u32 = 1;
+
+pub struct GammaControlManagerState {
+    gamma_controls: HashMap<ObjectId, ZwlrGammaControlV1>,
```

Comment:
```suggestion
    gamma_controls: HashMap<WlOutput, ZwlrGammaControlV1>,
```
Wayland objects conveniently implement Hash.